fiftyone/server/query.py

```diff
@@ -217,6 +218,9 @@ async def resolver(
             dataset.id = ObjectId()
             dataset.media_type = view.media_type
 
+        if dataset.media_type == fom.GROUP:
+            dataset.group_slice = view.group_slice
```

Comment:
Hmm maybe I should support for views having a custom `group_slice` that doesn't necessarily match their parent dataset's

Reply:
@benjaminpkane FYI just added support for setting `view.group_slice` independently of `dataset.group_slice` to this branch.

I guess the correct "synced" behavior is that changing the group slice in the App would immediately update both `session.dataset.group_slice` and `session.view.group_slice`.

Can you update the session logic as necessary so that:
- `session.dataset.group_slice` and `session.view.group_slice` matches the App's group slice setting? Since they should always match, it shouldn't be necessary to directly set `view.group_slice` here, so I guess maybe no code change is necessary here.
- Setting `session.view = view` will pull the view's group slice for the App setting. This will involve a slight state tweak

Reply:
Ok,, I believe these are addressed